spyder/plugins/ipythonconsole/comms/__init__.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) 2009- Spyder Kernels Contributors
+#
+# Licensed under the terms of the MIT License
+# (see spyder_kernels/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""
+API to communicate between the Spyder IDE and the Spyder kernel.
+It uses Jupyter Comms for messaging. The messages are sent by calling an
+arbitrary function, with the limitation that the arguments have to be
+picklable. If the function must return, the call must be blocking.
+
+In addition, the frontend can interrupt the kernel to process the message sent.
+This allows, for example, to set a breakpoint in pdb while the debugger is
+running. The message will only be delivered when the kernel is checking the
+event loop, or if pdb is waiting for an input.
+
+Example:
+
+On one side:
+
+    ```
+    def hello_str(msg):
+        print('Hello ' + msg + '!')
```

Comment:
Please change this to

      print('Hello ' + msg + '!')  # spyder: test-skip

to avoid an error with the test that verifies that no new print's are introduced in a PR.